lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1771,6 +1784,37 @@ instance Buildable PoolRetirementCertificate where
         <> " retiring at "
         <> build e
 
+-- | Represents an abstract notion of a certificate publication time.
+--
+-- Certificates published at later times take precedence over certificates
+-- published at earlier times.
+--
+data CertificatePublicationTime = CertificatePublicationTime
+    { slotId
+        :: SlotId
+    , slotInternalIndex
+        :: SlotInternalIndex
```

Comment:
Adding this newtype only clouds the meaning. Almost any integral type would be better.

Reply:
> Adding this newtype only clouds the meaning. Almost any integral type would be better.

Fair enough!

I've removed the `SlotInternalIndex` type by inlining it into `CertificatePublicationTime`, resulting in the following type:
```hs
data CertificatePublicationTime = CertificatePublicationTime
    { slotId
        :: SlotId
    , slotInternalIndex
        :: Word64
        -- ^ Indicates the relative position of a publication within a slot.
    }
    deriving (Eq, Generic, Ord, Show)
```
Fixed in c7e5da754889c5ba684fbd82fa3d899996ff0624.